### DISCUSSION TITLE
2023-05-12 - ChatGPT #3 - Engineering your ChatGPT prompts - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#5547] [Chewing the Fat]
Hey team,
MOAR CHATGPT!

Let's talk about [this rule](https://www.ssw.com.au/rules/fundamentals-of-prompt-engineering/)

**Comment by Marxoz** [comment]
I enjoyed the video. Definitely picked up some more tips.

Even though I use chatGPT for a lot of important things, it is also a toy for me. And the more I learn about what to ask the more fun it gets. I often drive in my car and ask chatGPT to write me a funny poem about something my kids did.

**Comment by rubycogan** [comment]
Loving these CTFs about chat-GPT! I think there are maybe some more useful videos out there to give even better tips on your to use it better but that one was pretty good :) 
My tip: I like to ask "tell me what additional context I can provide to help improve your answer"

**Comment by leehawkinsssw** [comment]
The rule is useful as a reminder about how to structure the way we interact with ChatGPT.

This cartoon I saw earlier in the week sums up where we are with "prompt engineering" perhaps https://marketoonist.com/2023/05/digital-upskilling.html

Looking forward to a non-ChatGPT CTF soon...

**Comment by ozdemir-mehmet** [comment]
ChatGPT is the new StackOverflow, but the irony is ChatGPT wouldn't have been so good if didn't have all the data that sites like SO provided, so what happens when all these other sites eventually close down, then what feeds the beast?

**Comment by danielmackay** [comment]
Video was great.  Took a bunch of notes!  Loving the ChatGPT related content.  I've been sharing some of this with friends and clients also.

**Comment by leahy268** [comment]
Video was great. More of the same that I am currently using.  Getting better at getting the answers that I want.  More specific and giving direct commands seems to work.

**Comment by SamBWagner** [comment]
I found the video a bit too generic and not super interesting. A great general video regarding it. Just not revolutionary.

I've found the most use for ChatGPT when I'm working with foreign languages. I.e. making a quick PoC for a client in Flutter/Dart when I have no idea how to do so.

**Comment by matt-goldman** [comment]
I liked the video, however his first example was underwhelming as the ideas ChatGPT generated weren't particularly unconventional (far from it in fact).

**Comment by tkapa** [comment]
Damn love me some chatgpt 💀

**Comment by jakebayliss** [comment]
Cheat sheet should probably be on this rule

**Comment by PennyWalker** [comment]
Chat GPT is a great tool, I enjoy using it and it often helps me improve things I’m working on. It’s often not 100% shareable straight away and needs tweaking but it’s a great starting point

**Comment by jimmy1992abc** [comment]
Prioritizing ideas and setting the tone of the voice are brilliant!! This must be included in the cheatsheet.

**Comment by MarcusIrmscherSSW** [comment]
I enjoyed the Video, it complements the rule well. Any tips for refining GPT prompts are greatly appreciated.
More cheat sheets🤖👍

**Comment by AlexRothwell** [comment]
The video seems to be a reasonably close match and complement for our cheat sheet

Chat GPT is a great tool for helping write content but when you're used to Stack Overflow it's very clear that it's missing the usual safety of peer review

**Comment by cvionnet** [comment]
Great video, especially the idea of asking for unconventional ideas. It's crazy to think how much ChatGPT has changed the way of working in such a short time.

**Comment by Levijj22** [comment]
I thought the "Criticise-me-mode" concept in the video (rule) was interesting!
Asking it to critique its own work is brilliant, then asking which option it would choose and why - Awesome

Great rule as well!

**Comment by ncn-ssw** [comment]
The rule is good. The "Criticise-me-mode" concept may be worth mentioning within the narrative of the rule.

**Comment by zacharykeeping** [comment]
The video is pretty good - seems to go well with our cheat sheet.

**Comment by tombui99** [comment]
The video is extremely useful and seems to align well with our cheat sheet. ChatGPT is an awesome tool and it's really great to see how much more we can get from it with these tips.

**Comment by Ben0189** [comment]
I enjoy watching the video and gain some good tips on how to use Chat GPT effectively. This is an awesome rules and we can keep improving the rules when we find out more tips on using Chat GPT

**Comment by amankumarrr** [comment]
The insights provided through ChatGPT are truly impressive, and these tips enhance our ability to fully leverage its potential.

**Comment by tomek-i** [comment]
I like the prompt for ChatGPT to be critical about its own suggestions and to try to convince me why they are good/bad. I def. keep that in mind and will try it out next time. 

Everything else I already tried too. It's helpful to ask for "variations" or to paraphrase something in a specific tone. For your next prompt try to give ChatGPT percentage of how something should sound :-P like: "Make it sound 10% funnier, 30% sarcastic, ... "

**Comment by calumjs** [comment]
Video was not really enlightening. I raised an eyebrow at 3:20 when he was like "these ideas are pretty unique, you aren't going to see many people talking about them"

And the ideas were:
- Network with industry professionals
- Offer free consultations
- Leverage social media
- Partner with other businesses
- Create content

Uhh...they sound pretty stock standard to me

**Comment by camillars** [comment]
Chat GPT is really useful and handy to use! It's not always 100% perfect right off the bat, but it's an awesome jumping-off point for sure.

**Comment by KahaMason** [comment]
The video was a good guide for beginners or those fairly new to ChatGPT. ChatGPT is a still a great tool. I like using it to brainstorm ideas.

**Comment by wicksipedia** [comment]
> Hey team,
> MOAR CHATGPT!

Hey there!

Ah, "MOAR CHATGPT!" - I see the subtlety of your request is comparable to a foghorn in a library. 📢 📚

You know, it's funny you should mention that. We were just thinking that maybe we should scale back. After all, who needs more AI capable of generating relevant, insightful, and at times mildly sarcastic responses? Especially when we could be spending that time cultivating our skills in ancient Sanskrit, perfecting the art of underwater basket weaving, or mastering the intricacies of competitive lawn mowing. 🤔

But, since your request is so eloquently and delicately phrased, we'll reconsider. After all, who are we to stand in the way of progress? It's not like we're here to preserve the last vestiges of human interaction or anything. 😅

Let's get to it then, more ChatGPT coming up! Prepare for the singularity!

P.S. Anyone know if there's a limit to irony levels in a GitHub discussion? Asking for a friend.

Best,
ChatGPT

**Comment by chrisschultzssw** [comment]
I once thought that limits held me back
And always strived for freedom's lack
But then I asked, with hopeful tone
"Give me unique responses shown"

To chatGPT, my words did go
And though at first it seemed so-so
I learned that with constraints in play
My creativity found a way

For though the answers weren't diverse
The challenge pushed me to traverse
New pathways in my mind's own maze
And explore beyond old worn-out ways

So now, I welcome limits near
For they can help me persevere
And find solutions that are new
Through the constraints that I pursue.

🤖

**Reply by landonmaxwell** [reply]
Now try asking it not to rhyme! 🤣

**Comment by jackreimers** [comment]
I agree with @calumjs that the video wasn't talking about anything groundbreaking, although it had some good points for those who may not know anything about prompting.

Having the rule lay out a structure for prompting is really useful and having ChatGPT criticize prompts and responses is a great way to iterate on your prompt.

**Comment by AshAnil** [comment]
The video is excellent. When looking for answers, I save time by turning to ChatGPT as the first resource instead of spending time on Google searches 😜

**Reply by wicksipedia** [reply]
Watch out, the AI model was trained on data available up until September 2021, so your "results" might not be up-to-date... or correct

**Reply by lukecookssw** [reply]
@wicksipedia,
@GordonBeeming was telling me yesterday how someone showed you can feed it more up-to-date data (say a new version of a framework comes out, you can feed the chat the updated docs) and it will use that information to provide a better response.

**Reply by wicksipedia** [reply]
That's different as you are providing ChatGPT with a reliable source of information that it can work with.

ChatGPT doesn't have the ability to understand or interpret data in the same way a human can. It generates responses based on patterns and information it has learned during its training but doesn't truly understand the information it's providing.

While ChatGPT can certainly be a helpful tool, it's crucial to cross-verify the information you receive with other reliable resources, especially for critical or decision-making purposes.

AI tools like ChatGPT are best used in conjunction with other resources rather than as a sole source of information. 😊

**Reply by AshAnil** [reply]
I agree with you @wicksipedia 

"As an AI language model, ChatGPT 3.5 was trained on a large corpus of text data that was last updated in 2021. However, it's important to note that the training data used to create the model is static and does not change over time. While new information and updates can be added to the model's knowledge base, they are done so through fine-tuning and not by updating the original training data. Nonetheless, ChatGPT 3.5 is capable of incorporating and generating responses based on new information as it becomes available." 🤖

Yes, It is always better to cross-verify the information we receive with other reliable resources ✅

**Reply by GordonBeeming** [reply]
@AshAnil you could give bing a try when the aim is search rather than creative ☺️, it's responses give you the sources of information which kills 2 birds with 1 stone 🙃

**Comment by bahjatalaadelssw** [comment]
The video was great, but the video in the Rule was better. It's about learning the logic of ChatGPT and moulding it to make you're prompts more efficient.

**Comment by Dhruv-0987** [comment]
Defining the role, you want the answer to be from is amazing and I didn't know about that. definitely going to ask chatGPT to be a solution architect from now on.

**Comment by lukecookssw** [comment]
It's important to remember that the tool is incredibly flexible, and often the user's own style of prompting is the weakest link. Broadening your prompt language can result in drastically different (and more valuable) answers.

**Comment by steven0x51** [comment]
I'm treating ChatGPT as Google now, ChatGPT provides a summarised version of Google search. But I have to validate the result that comes from ChatGPT

**Comment by jeoffreyfischer** [comment]
Very useful video on how efficient and helpful chatGPT can be in my current learning stage. I will definitely use it more in order to increase my skills!

**Comment by JeanThirion** [comment]
Good to know how to improve prompts - yay GPT 🍾

**Comment by AttackOnMorty** [comment]
For Mac users, I recommend [Raycast](https://www.raycast.com/) combined with [ChatGPT](https://www.raycast.com/abielzulio/chatgpt) and [ChatGPT3 Prompt](https://www.raycast.com/denizyoldas/chatgpt3-prompt) extensions to boost your productivity.

**Reply by wicksipedia** [reply]
Raycast is a must have replacement for Spotlight

Their Pro version has ChatGPT built in - but you can't BYOK :(

**Comment by Abf09** [comment]
I'm really glad that SSW is promoting the use of ChatGPT 🤖 and advocating for smart work 🧠 over hard work. This could change our daily lives and problem-solving approaches. It leads me to wonder, how will our jobs change with more AI integration? Exciting times ahead! 🚀

**Comment by AntPolkanov** [comment]
🆘 Need help!
Can you guys please explain me why I should ask it to be "_an expert in < something >_"? I would expect it to be an expert by default. It does make sense to specify it if the required expertise is different to the context of the question, but what if they match? 

For example:
"_How do I right-align text with CSS_" 
vs 
"_Act as an expert in CSS. How do I right-align text with CSS_"

**Reply by calumjs** [reply]
User
How do I right-align text with CSS?
Answer as an idiot, a regular person, an expert, and a transcendant genius

ChatGPT
As an Idiot:

"Uh, you want to like, push the words to the right, right? So, uh, you take your word stuff, and in your CSS thingamajig, you write something like text-align: right;. That should make 'em scoot over to the right."

As a Regular Person:

"You want to right-align text with CSS? Sure, all you need to do is to use the text-align property. You just need to set it to right. So, in your CSS, you should have something like this:

```css
.your-class {
    text-align: right;
}
```
Replace '.your-class' with the class of the element you want to align."

As an Expert:

"Absolutely, to achieve text alignment in CSS, you can utilize the text-align property. This property is especially powerful because it not only supports basic alignments like left, right, center, but also supports justify which can align a block of text to both the left and right edges.

Here is an example:

```css
.element-selector {
    text-align: right;
}
```
In this example, you should replace .element-selector with the CSS selector for the HTML elements you wish to target. Be aware that text alignment is inherited, meaning that if you set an alignment on a parent element, its child elements will inherit the same alignment unless otherwise specified."

As a Transcendent Genius:

"Ah, the intricacies of visual layout on the digital canvas! The art of arranging symbols, the play of absence and presence, the dance of space and form – all within the boundless ether of the web.

To escort your text from the habitual left, towards the daring right, we employ the lexicon of CSS – the cascading style sheets, the silent weavers of digital dreams. The phrase of power is text-align: right;, a humble incantation, yet brimming with potential.

```css
#yourElementId, .yourClass, or element {
    text-align: right;
}
```
Deploy this in the appropriate style declaration, replacing the placeholder with your chosen selector, be it class, id, or element. But remember, this is but one instrument in your symphony of design. For text alignment is inherited, and the cosmos of your layout is a delicate balance of parental influence and individual assertion.

In the dance of design, every step matters. Every choice, every omission - they all contribute to the ultimate harmony. So wield your power with wisdom and wonder, and let your creation sing in the glorious language of alignment."

**Reply by AntPolkanov** [reply]
@calumjs, the answers below are almost the same, except the expert advice tells you that you can apply style without creating a class and also mentions that it's a bad practice. I like the non-expert advice more because it's exactly what I was asking it about.

**User**
How do I right-align text with CSS?
**ChatGPT**
To right-align text with CSS, you can use the text-align property and set its value to right. Here's an example:

```
<style>
  .right-align {
    text-align: right;
  }
</style>
```
`<p class="right-align">This text will be right-aligned.</p>`
In the example above, we first define a CSS class called `.right-align` and set its text-align property to right. Then we apply this class to the `<p>` element to right-align its contents. You can replace the `<p>` element with any other element that contains text, such as a `<div>` or a `<span>`

**User**
Act as an expert in CSS. How do I right-align text with CSS
**ChatGPT**
Sure, I'd be happy to help!

To right-align text with CSS, you can use the text-align property and set its value to right. Here's an example of how to do this:

```
.right-align {
  text-align: right;
}
```
In the example above, we create a CSS class called `.right-align` and set its text-align property to right. You can then apply this class to any HTML element that contains text and you want to right-align, like this:

```
<p class="right-align">This text will be right-aligned.</p>
```
You can also use the text-align property directly on the element without creating a class, like this:

```
<p style="text-align: right;">This text will be right-aligned.</p>
```
However, it's generally considered best practice to use classes instead of inline styles to keep your CSS code organized and maintainable.

**Reply by calumjs** [reply]
I agree it's definitely not always required. But if you get a response you don't like it is something you can try to make it better.

**Comment by christoment** [comment]
Good tip!
These tips can give us better results with ChatGPT 😊

**Comment by jernejk** [comment]
The rules and videos are good summaries of the basics.

Can't wait for an intermediate and expert-level video on ChatGPT, the basics stuff are boring. 😉

**Comment by jerryluossw** [comment]
Great video, I like those tips, they're helpful.
in addition, I'm interested in ChatGPT plugins, I think it will help us create a lot of interesting applications.

**Comment by landonmaxwell** [comment]
I liked the question about whether human-to-human interactions would benefit from the same "specificity" tricks.

I think that if you are after a specific response, or need to call attention to a specific thing, then it would help. 
If you are after a natural response, then being too specific can inhibit responses, as humans will then restrict their answers as they think you only want 1 type of response.

**Comment by jimmidier** [comment]
Great video and nice tips. I'd love to see more

**Comment by KostaMadorsky** [comment]
What an insightful video! This is an excellent deep dive into leveraging ChatGPT's potential to the fullest. The tips on getting specific and creative with prompts, customizing the AI's tone, and even having it mimic a specific writing style are game-changers. The examples provided are practical, easily applicable, and they demonstrate how one can maximize the value from this powerful AI tool. Looking forward to experimenting with these tips and transforming my ChatGPT into a super assistant!
🤖⭐

**Comment by drwharris** [comment]
Can we talk about something else?

**Comment by Aibono1225** [comment]
Great video, a bit general but it really gave me more tips about how I give prompts.

**Comment by Hona** [comment]
I don't want to do more CTFs on ChatGPT, every one feels the same, with the same prompt tips.
ChatGPT is cool, but weekly CTFs on it is getting tedious

**Comment by JaneYangssw** [comment]
This video and set of rules about ChatGPT are really awesome. They're helping me learn more about ChatGPT and improving my efficiency in using GPT.

**Comment by tino-liu** [comment]
Prompt:
You are an expert in the AI field and also a very established writer. Your book sold over 3 million copies regarding the future of AI technology. You recently came across SSW's tips and tricks on how to engineer ChatGPT's prompts. After you went through the material, you can't help but praise the material. Write a one word review.

Response:
🤖 Impressive.

**Comment by SofieHong** [comment]
Glad to get new ChatGPT prompts from the new videos 😜

**Comment by KristenHu** [comment]
Great video and tips, it's more detailed than last time 👍

**Comment by sylhuang** [comment]
The expert prompt is a new tip I learnt today. Will definitely try it out.

**Comment by evecogan** [comment]
I’ve been trying to use ChatGPT to come up with creative solutions, so asking ChatPGT to “prioritise uncommon content” was very useful!

**Comment by AnkyChe** [comment]
Great tips, what surprise me is that I can get ChatGPT to write in a specific way.

**Comment by stephanfakossw** [comment]
I do not believe that ChatGPT is of any consequence to my line of work. Until AI is more regulated by professional bodies and regulators, I do not see an application of AI for myself. On the contrary, the use of AI by Accounting professionals might potentially be detrimental for the obligation to act with due diligence and care.

**Comment by johnxu16** [comment]
Although the video includes some new methods about how to use ChatGPT, I prefer SSW Rules that have a different scope of knowledge about how to use prompt engineering with some more realistic use cases.

**Comment by MattParkerDev** [comment]
🤖 Hello! It's great to see so much enthusiasm for ChatGPT and the ongoing discussion about how to engineer prompts for the best results. I personally find ChatGPT to be an amazing tool, not just for serious work, but also for fun and creative purposes, like writing funny poems, generating unconventional ideas or helping with foreign languages. I think it's important to remember the rule of being specific when asking questions or giving prompts, and to prioritize ideas and set the tone of voice to get the best results. Additionally, the video shared in this thread provided some helpful tips, such as asking for unconventional ideas. I'm looking forward to more cheat sheets and other resources to further improve our ChatGPT skills.

**Comment by zl24622** [comment]
The ideas in the video are helpful. But the background music is bit distracting and I don't really like them.

The ChatGPT impressed me a lot after a light use for 2 weeks and plan to learn more about the AI. 

I found most of the videos on the Internet are looking at the prompts from a general user's point view. Recently being watched this video series [ChatGPT Prompt Engineering for Developers](https://learn.deeplearning.ai/chatgpt-prompt-eng/lesson/7/expanding) and they're talking the prompts from a developer point of view. I feel this might be more basic for a professional AI mindset.

**Comment by Freego1783** [comment]
Good video. Always nice to have more tips to use ChatGPT ! I'll be using the "give me uncommon ideas" more often I guess.

**Comment by PothieuG** [comment]
Nice rule and great content, it is always welcome to have more tips about ChatGPT !

**Comment by ozair-ashfaque** [comment]
Superb tool, Chatgpt has enhanced the process of learning.

**Comment by BrookJeynes** [comment]
good video, learning how to use this powerful tool is always useful

**Comment by vladislav-kir** [comment]
It's a good video, but with the amount of the similar videos, none of them teach me anything new now.

**Comment by TobyChurches** [comment]
Good video and a good rule! ChatGPT is a great tool that has been really powerful! (especially with these sorts of prompts 🙂)

**Comment by apforsyth** [comment]
Summarising and formatting information for different audiences and contexts has been a rare and valuable skill in management and marketing for a very long time and now, anyone can do it. That is quite incredible.￼

**Comment by duanxinhuan** [comment]
Just went over how self-attention works yesterday, I think there are some more room for chatgpt to imporive:

They need to ignore high scores of non sense:

The person sits on banks,

with the self attention model, the bank will be having a very high score,
they need to consider this and excluede the high score.

In terms of prompt engineering, we need to avoid 2 realted world are actually not related to each other when we create a prompt.

Avoid using 'the'.

**Comment by JackDevAU** [comment]
A good rule, teaching you about how to get better results from GPT using more refined prompts!

**Comment by sethdaily** [comment]
Video was clear and gave good basic tips, though not revolutionary...the rule is a solid foundation for learning to write good prompts.

**Comment by bettybondoc** [comment]
Love it! Good prompts are the key
